### PR TITLE
DDF-1777 Added a max option for replicate and migrate catalog-core-commands

### PR DESCRIPTION
**Pull request by bdeining**
@millerw8 is hero
@dcruver 
@shane-voisard 
@kcwire

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/446" target="_blank">ReviewNinja

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java [hunk 1]
```diff
@@ -106,32 +108,37 @@
...
-        List<Metacard> queryMetacards = null;
+        List<Metacard> queryMetacards;
         queryMetacards = query(queryFacade, startIndex, filter);
 
         if (queryMetacards == null || queryMetacards.isEmpty()) {
             return 0;
         }
 
+        // If maxMetacards is set, restrict the amount of ingested metacards to the maxMetacards
+        if (maxMetacards > 0 && ((queryMetacards.size() + successCount.get()) > maxMetacards)) {
```
Comment (millerw8) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: Why can't we just set the pagesize of the query to maxMetacards when maxMetacards - successCount > batchSize.

Seems like if I decided I wanted to set a scheduled command to pull back at most 1 records from a remote site every 5 minutes, and I didn't set the batch size, I would end uprequesting up to 1000 records every 5 minutes and then dropping all but 1 on the ground

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java [hunk 2]
```diff
@@ -106,32 +108,37 @@
                     + "\tComplex:   search --cql \"title like 'some text' AND modified before 2012-09-01T12:30:00Z\"")
     String cqlFilter = null;
 
-    private List<Metacard> failedMetacards = Collections
-            .synchronizedList(new ArrayList<Metacard>());
+    @Option(name = "--max", required = false, aliases = {
```
Comment (millerw8) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: Can we make this --maximum with an alias of -max, or maybe --maximumMetacards and -mm, or maybe --maximum and -m and change multiThreaded alias to -M.  Our usage and aliases seem quite horrible, and out of order.  I will comment on the others in another ticket.  But it seems the general convention is --<complete-name> and -<shortened-name or initial>.  However, you look at our help file and it is all over the place, including stuff like -lastmonths and -month (note no double dash and -e and -enddate, where you have no double dash and the shorter one is the name, not the alias.  It ends up looking very amateur when printing it out from command line

Comment (millerw8) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: !fix

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java (hunk 1 above)
Comment (millerw8) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: !fix

Comment (bdeining) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: !fixed

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java (hunk 2 above)
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: !fixed

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java (hunk 1 above)
Comment (millerw8) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java]: Looks good in your branch, could you add it to the pull request and I will give test it some more and merge it

**Pull request merged**